WB-5

duncte123 opened the pull request:
Create the command system and add two commands

duncte123 committed:
Create command system

Create the command system and add two commands

duncte123 committed:
Fix codacy issues

duncte123 committed:
Add unit test

duncte123 committed:
Fix more codacy issues

duncte123 committed:
Fix more codacy issues

Please let them stop.

duncte123 committed:
Fix remaining codacy issues for the command system

duncte123 committed:
Fix additional codacy issues for command system

duncte123 committed:
Fix more issues for codacy

Remove  fianal variables in WatsonCommandTest
Turn \n into a character and not a string

duncte123 committed:
Fix more issues for codacy

Remove final variables in WatsonCommandTest, Turn \n into a character and not a string

duncte123 committed:
Merge branch 'WB-5' of https://github.com/Discord-Dutch-HypeSquad/Watson-Bot into WB-5

duncte123 committed:
Finalise fixes for codacy issues

duncte123 committed:
Update gitignore to remove idea files

duncte123 committed:
Merge branch 'dev' into WB-5

kevzz1994 commented on src/main/java/io/hypesquad/watsonbot/event/WatsonEventListener.java:
```diff
@@ -17,24 +17,53 @@
             return;
+        }
+
+        //Handle command
```
This can be done much easier. Find a better and more logical way.

duncte123 committed:
Remove message in WatsonCommandTest

duncte123 committed:
Add an equals method to the command class

Make the WatsonCommand class abstract. Make WatsonExampleCommand output the arguments that you give it

duncte123 committed:
Correct Codacy issues

duncte123 committed:
Revers ranaming command classes

cascer1 committed:
Fix equals method for AbstractWatsonCommand, clean up gitignore

cascer1 committed:
Add final modifier to AbstractWatsonCommand.equals argument

duncte123 committed:
Make command parsing better

duncte123 committed:
Remove unused import

cascer1 committed:
Modify AbstractWatsonCommand javadoc

kevzz1994 commented on src/main/java/io/hypesquad/watsonbot/commands/AbstractWatsonCommand.java:
```diff
@@ -32,7 +32,7 @@
      * @param args  the arguments from the command
      * @param event the MessageReceivedEvent from D4J
-     * @return true or false depending on some checks that we run
+     * @return ture if checks pass, false otherwise
```
Quite the typo you have there.

cascer1 replied:
fffffff

cascer1 committed:
Fix typo in AbstractWatsonCommand javadoc

kevzz1994 commented on src/main/java/io/hypesquad/watsonbot/commands/WatsonHelpCommand.java:
````diff
@@ -0,0 +1,58 @@
+    @Override
+    public void executeCommand(final String[] args, final MessageReceivedEvent event) {
+
+        final StringBuilder stringBuilder = new StringBuilder(80).append("```\n");
````
Why ``80``?

duncte123 committed:
Remove capacity from stringbuilder

duncte123 committed:
Lower capacity to 20

kevzz1994 commented on src/main/java/io/hypesquad/watsonbot/commands/WatsonHelpCommand.java:
````diff
@@ -0,0 +1,58 @@
+    @Override
+    public void executeCommand(final String[] args, final MessageReceivedEvent event) {
+
+        final StringBuilder stringBuilder = new StringBuilder(20).append("```\n");
````
``new StringBuilder(20)``, 20 here is a magic number. Refactor to a constant.

duncte123 committed:
Replace magic number with constant

duncte123 committed:
Shorten variable name for MAX_COMMAND_HELPMESSAGE_LENGTH

duncte123 committed:
Add override for hashCode

kevzz1994 commented on src/main/java/io/hypesquad/watsonbot/commands/AbstractWatsonCommand.java:
```diff
@@ -0,0 +1,80 @@
+
+    @Override
+    public int hashCode() {
+        return 1;
```
Hashcode should be calculated differently.

duncte123 committed:
Change hashCode method
